pandas/tests/io/test_sql.py

```diff
@@ -2698,6 +2700,64 @@ def test_drop_table(conn, request):
         assert not insp.has_table("temp_frame")
 
 
+@pytest.mark.parametrize("conn", all_connectable)
+def test_delete_rows_success(conn, test_frame1, request):
+    table_name = "temp_frame"
+    conn = request.getfixturevalue(conn)
+    pandasSQL = pandasSQL_builder(conn)
+
+    with pandasSQL.run_transaction():
+        assert pandasSQL.to_sql(test_frame1, table_name) == test_frame1.shape[0]
+
+    with pandasSQL.run_transaction():
+        assert pandasSQL.delete_rows(table_name) is None
+
+    assert count_rows(conn, table_name) == 0
+    assert pandasSQL.has_table("temp_frame")
+
+
+@pytest.mark.parametrize("conn", all_connectable)
+def test_delete_rows_is_atomic(conn, request):
+    adbc_driver_manager = pytest.importorskip("adbc_driver_manager")
+    sqlalchemy = pytest.importorskip("sqlalchemy")
+
+    if "sqlite" in conn:
```

Comment:
Can you either rework this test so we can test SQLite or give it its own dedicated test case? I think right now with this xfail it would be possible to SQLite to actually NOT be atomic, but that would get lost in the xfail that we assume deals with the data types